src/windows.rs

```diff
@@ -1,59 +1,30 @@
 //! Implementation for Windows
 use crate::Error;
-use core::{ffi::c_void, mem::MaybeUninit, num::NonZeroU32, ptr};
+use core::mem::MaybeUninit;
 
-const BCRYPT_USE_SYSTEM_PREFERRED_RNG: u32 = 0x00000002;
+type HMODULE = isize;
+type BOOL = i32;
+const TRUE: BOOL = 1;
 
-#[link(name = "bcrypt")]
-extern "system" {
-    fn BCryptGenRandom(
-        hAlgorithm: *mut c_void,
-        pBuffer: *mut u8,
-        cbBuffer: u32,
-        dwFlags: u32,
-    ) -> u32;
-}
+type ProcessPrng = unsafe extern "system" fn(*mut u8, usize) -> BOOL;
 
-// Forbidden when targetting UWP
-#[cfg(not(target_vendor = "uwp"))]
-#[link(name = "advapi32")]
+#[link(name = "kernel32")]
 extern "system" {
-    #[link_name = "SystemFunction036"]
-    fn RtlGenRandom(RandomBuffer: *mut c_void, RandomBufferLength: u32) -> u8;
+    fn LoadLibraryA(libfilename: *const u8) -> HMODULE;
```

Comment:
Chromium and some other projects specifically uses `LoadLibraryW`, so it would be good to use it instead of `LoadLibraryA`.

More generally, this won't work inside many sandboxes, including Chromium's.

See https://source.chromium.org/chromium/chromium/src/+/main:sandbox/policy/win/sandbox_warmup.cc;bpv=1;bpt=1. In order to support sandboxes, `getrandom` should have a similar "call this before enabling the sandbox" API that does the bare minimum (LoadLibrary, and then store the function pointer, without actualy calling `ProcessPrng`).

IIRC, on Windows there is a way to tell the linker to generate metadata that says "we will delay-load bcryptprimitives.dll from system32". using `/delayload:bcryptprimitives.dll` or similar. I can't remember why, but in other projects I've worked on, we specifically had to ensure that we did this for every delay-loaded library. I think we had to do this because there is something (internal to those projects? anti-virus in general?) that freaks out when LoadLibrary is used on something that isn't mentioned in that metadata.

Reply:
> IIRC, on Windows there is a way to tell the linker to generate metadata that says "we will delay-load bcryptprimitives.dll from system32". using /delayload:bcryptprimitives.dll or similar. I can't remember why, but in other projects I've worked on, we specifically had to ensure that we did this for every delay-loaded library. I think we had to do this because there is something (internal to those projects? anti-virus in general?) that freaks out when LoadLibrary is used on something that isn't mentioned in that metadata.

Also, these were Rust-based applications that used (indirectly) `getrandom`, though it has been a while since I've worked on them.

Reply:
We no longer dynamically choose the implementation on Windows, so resolving this.